### DISCUSSION TITLE
Faker set locale

### DISCUSSION THREAD

**Discussion by clabnet** [adonisjs/core#3447] [Help] [answered]
Hi, 

how to  setLocale to faker on factories Adonis 5 ? 

```
import 'faker'
import * as faker from 'faker/locale/it'

faker.setLocale('it')    <<<--- this has no effect

export const PartnerFactory = Factory.define(Partner, ({ faker }) => {
...
}
```

**Comment by MaximeMRF** [comment]
Do not use a external package for this, you can use [Adonis factories](https://docs.adonisjs.com/guides/models/factories#document)

**Reply by clabnet** [reply]
> Do not use a external package for this, you can use [Adonis factories](https://docs.adonisjs.com/guides/models/factories#document)

In effect my question is related to Factories.

**Comment by Julien-R44** [comment, marked as answer]
You can do it like this : 

```ts
export const AccountFactory = Factory.define(Account, ({ faker }) => {
  faker.setLocale('it')
  ...
  return {}
})
```

Looking at the lucid source code, there doesn't seem to be a way to globally define faker localization at the moment. That said, the feature could be added quite easily, if you have the time to do a PR !

**Reply by clabnet** [reply]
It work fine.  Thanks @Julien-R44 

```
import Factory from '@ioc:Adonis/Lucid/Factory'
import Partner from 'App/Models/Partner'
import { DateTime } from 'luxon'

export const PartnerFactory = Factory.define(Partner, ({ faker }) => {
  faker.setLocale('it')
  return {
    company_name: faker.company.companyName(),
    email: faker.internet.email(),
    created_by: faker.datatype.number({ min: 1, max: 2 }),
    createdAt: DateTime.fromJSDate(faker.date.recent()),
  }
}).build()

```
